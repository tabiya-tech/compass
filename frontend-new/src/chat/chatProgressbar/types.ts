export enum ConversationPhase {
  INITIALIZING = "INITIALIZING",
  INTRO = "INTRO",
  COLLECT_EXPERIENCES = "COLLECT_EXPERIENCES",
  DIVE_IN = "DIVE_IN",
  ENDED = "ENDED",
  UNKNOWN = "UNKNOWN"
}

export type CurrentPhase = {
  percentage: number
  phase: ConversationPhase
}

export const defaultCurrentPhase = {
  phase: ConversationPhase.INITIALIZING,
  percentage: 0
}
