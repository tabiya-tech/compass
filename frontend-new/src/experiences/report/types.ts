import { Experience } from "src/experiences/experiencesDrawer/experienceService/experiences.types";

export type ReportProps = {
  name: string;
  email: string;
  phone: string;
  address: string;
  experiences: Experience[];
  conversationConductedAt: string | null;
};

export enum ExportFormat {
  PDF = "pdf",
  DOCX = "docx",
}

export interface IReportFormatProvider {
  download: () => void;
}
