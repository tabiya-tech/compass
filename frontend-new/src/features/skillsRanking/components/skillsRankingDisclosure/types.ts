export const jobSeekerComparisonLabels = ["LOWEST", "SECOND LOWEST", "MIDDLE", "SECOND HIGHEST", "HIGHEST"]