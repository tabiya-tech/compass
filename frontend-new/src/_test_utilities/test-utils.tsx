// Based on https://testing-library.com/docs/react-testing-library/setup/
import React, { ReactElement, ReactNode } from "react";
import { render, renderHook, RenderHookOptions, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import applicationTheme, { ThemeMode } from "src/theme/applicationTheme/applicationTheme";

// Import the Firebase mock utilities
import "src/_test_utilities/firebaseMock";
import SnackbarProvider from "src/theme/SnackbarProvider/SnackbarProvider";
import { IsOnlineProvider } from "src/app/isOnlineProvider/IsOnlineProvider";
jest.mock("firebase/compat/app", () => {
  return {
    initializeApp: jest.fn(),
    auth: jest.fn().mockReturnValue({
      signOut: jest.fn(),
      currentUser: {
        getIdToken: jest.fn(),
        getTokenResult: jest.fn(),
      },
      onAuthStateChanged: jest.fn().mockReturnValue(jest.fn()),
      signInWithPopup: jest.fn(),
      GoogleAuthProvider: { PROVIDER_ID: "google.com" },
    }),
  };
});

jest.mock("firebaseui", () => {
  return {
    auth: {
      AuthUI: {
        start: jest.fn(),
        getInstance: jest.fn().mockReturnValue({
          start: jest.fn(),
          reset: jest.fn(),
        }),
        reset: jest.fn(),
      },
    },
  };
});

jest.mock("@sentry/react", () => ({
  withSentry: (Component: any) => Component,
  getFeedback: jest.fn(),
  withProfiler: (Component: any) => Component,
  wrapCreateBrowserRouter: (Component: any) => Component,
}));

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <IsOnlineProvider>
      <ThemeProvider theme={applicationTheme(ThemeMode.LIGHT)}>
        <SnackbarProvider>{children}</SnackbarProvider>
      </ThemeProvider>
    </IsOnlineProvider>
  );
};

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, "wrapper">) =>
  render(ui, { wrapper: AllTheProviders, ...options });

function customRenderHook<T>(
  hook: () => T,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { wrapper?: React.ComponentType }
) {
  const CombinedWrapper = ({ children }: { children: ReactNode }) => {
    const Wrapper = options?.wrapper ?? React.Fragment;
    return (
      <AllTheProviders>
        <Wrapper>{children}</Wrapper>
      </AllTheProviders>
    );
  };

  return renderHook(hook, { wrapper: CombinedWrapper, ...options });
}

export * from "@testing-library/react";
export { customRender as render };
export { customRenderHook as renderHook };
