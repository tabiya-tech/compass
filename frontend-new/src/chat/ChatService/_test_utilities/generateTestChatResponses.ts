import { ConversationMessage, ConversationMessageSender } from "src/chat/ChatService/ChatService.types";

export const generateTestChatResponses = (): ConversationMessage[] => {
  return [
    {
      message: "Hello! How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "I can help you with that. What is your name?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
  ];
};

export const generateTestHistory = (): ConversationMessage[] => {
  return [
    {
      message: "Hello! How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "I can help you with that. What is your name?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
    {
      message: "My name is John.",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.USER,
    },
    {
      message: "Nice to meet you, John. How can I help you today?",
      sent_at: new Date().toISOString(),
      sender: ConversationMessageSender.COMPASS,
    },
  ];
};
