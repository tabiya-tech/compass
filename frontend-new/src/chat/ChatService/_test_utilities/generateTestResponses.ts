import { LastMessage, RootObject } from "src/chat/ChatService/ChatService.types";

export const generateRootObjectResponse = (): RootObject => ({
  last: {
    message_for_user: "Hello",
    finished: false,
    agent_type: "WelcomeAgent",
    reasoning: "Initial greeting",
    agent_response_time_in_sec: 1.0,
    llm_stats: [
      {
        error: "",
        prompt_token_count: 50,
        response_token_count: 10,
        response_time_in_sec: 1.0,
      },
    ],
  },
  conversation_context: {
    all_history: {
      turns: [
        {
          index: 1,
          input: {
            message: "Hello",
          },
          output: {
            message_for_user: "Hi there!",
            finished: false,
            agent_type: "WelcomeAgent",
            reasoning: "Greeting response",
            agent_response_time_in_sec: 0.5,
            llm_stats: [
              {
                error: "",
                prompt_token_count: 30,
                response_token_count: 10,
                response_time_in_sec: 0.5,
              },
            ],
          },
        },
      ],
    },
    history: {
      turns: [],
    },
    summary: "",
  },
});

export const generateLastMessageResponse = (): LastMessage => ({
  message_for_user: "Hi there!",
  finished: false,
  agent_type: "WelcomeAgent",
  reasoning: "Greeting response",
  agent_response_time_in_sec: 1.0,
  llm_stats: [
    {
      error: "",
      prompt_token_count: 30,
      response_token_count: 10,
      response_time_in_sec: 0.5,
    },
  ],
});
