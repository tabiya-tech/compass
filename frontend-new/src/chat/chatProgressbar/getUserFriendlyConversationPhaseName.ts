import { ConversationPhase, CurrentPhase } from "./types";
import i18n from "src/i18n/i18n";
// Compute translations at call time so they react to language changes and avoid module-scope resolution.
export const getUserFriendlyPhaseNames = (): Record<ConversationPhase, string> => ({
  [ConversationPhase.INITIALIZING]: i18n.t("phase_initializing"),
  [ConversationPhase.INTRO]: i18n.t("phase_intro"),
  [ConversationPhase.COLLECT_EXPERIENCES]: i18n.t("phase_collecting"),
  [ConversationPhase.DIVE_IN]: i18n.t("phase_exploring"),
  [ConversationPhase.ENDED]: i18n.t("phase_finished"),
  [ConversationPhase.UNKNOWN]: i18n.t("phase_unknown"),
});

/**
 * Get a user-friendly name for the conversation phase
 * @param phase {CurrentPhase} - The conversation phase object.
 * @returns {string} User-friendly, translated name (with progress if applicable)
 */
export function getUserFriendlyConversationPhaseName(phase: CurrentPhase): string {
  const USER_FRIENDLY_PHASE_NAMES = getUserFriendlyPhaseNames();
  if (phase.phase === ConversationPhase.COLLECT_EXPERIENCES) {
    return `${USER_FRIENDLY_PHASE_NAMES[ConversationPhase.COLLECT_EXPERIENCES]}: ${phase.current}/${phase.total} ${i18n.t("work_types")}`;
  }

  if (phase.phase === ConversationPhase.DIVE_IN) {
    return `${USER_FRIENDLY_PHASE_NAMES[ConversationPhase.DIVE_IN]}: ${phase.current}/${phase.total} ${i18n.t("experiences")}`;
  }

  return USER_FRIENDLY_PHASE_NAMES[phase.phase] || USER_FRIENDLY_PHASE_NAMES[ConversationPhase.UNKNOWN];
}