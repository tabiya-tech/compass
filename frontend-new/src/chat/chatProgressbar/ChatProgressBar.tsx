import React from "react";
import { CurrentPhase } from "./types";
import { motion } from "framer-motion";
import { Box, useTheme, Typography } from "@mui/material";
import { getUserFriendlyConversationPhaseName } from "./getUserFriendlyConversationPhaseName";

const uniqueId = "2a76494f-351d-409d-ba58-e1b2cfaf2a53";

export const DATA_TEST_ID = {
  CONTAINER: `chat-progress-bar-container-${uniqueId}`,
  PROGRESS_BAR: `chat-progress-bar-${uniqueId}`,
  PROGRESS_BAR_PHASE_TEXT: `chat-progress-bar-text-${uniqueId}`,
  PROGRESS_BAR_LABEL: `chat-progress-bar-label-${uniqueId}`,
};

const ChatProgressBar: React.FC<CurrentPhase> = (currentPhase) => {
  const theme = useTheme();
  const percentageInText = `${currentPhase.percentage}%`;
  const userFriendlyConversationPhaseText = getUserFriendlyConversationPhaseName(currentPhase);

  return (
    <Box sx={{
      display: "flex",
      flexDirection: "column",
      gap: theme.fixedSpacing(theme.tabiyaSpacing.xs),
      // to match the width of the message list container
      [theme.breakpoints.up("md")]: {
        width: "60%",
        margin: "auto",
      },
    }}>
      <Box
        data-testid={DATA_TEST_ID.CONTAINER}
        sx={{
          overflow: "hidden",
          fontSize: theme.typography.overline.fontSize,
          width: "100%",
          backgroundColor: theme.palette.common.white,
          borderRadius: theme.rounding(theme.tabiyaRounding.xs),
          border: `1px solid ${theme.palette.grey[300]}`,
          display: "grid",
          placeItems: "center start",
        }}
      >
        <motion.div
          style={{
            backgroundColor: theme.palette.primary.main,
            height: "100%",gridRow: 1,
            borderRadius: theme.rounding(theme.tabiyaRounding.xs),
            gridColumn: 1
          }}
          initial={{
            width: 0,
          }}
          animate={{
            width: percentageInText,
          }}
          data-testid={DATA_TEST_ID.PROGRESS_BAR}
          transition={{
            duration: 0.5,
            ease: "easeOut",
          }}
        />
        <Box
          sx={{
            display: "flex",
            width: "100%",
            justifyContent: "space-between",
            alignItems: "center",
            paddingX: theme.fixedSpacing(theme.tabiyaSpacing.sm),
            paddingY: theme.fixedSpacing(theme.tabiyaSpacing.xxs),
            gridRow: 1,
            gridColumn: 1
          }}
        >
          <Typography variant="progressBarText" lineHeight={1} data-testid={DATA_TEST_ID.PROGRESS_BAR_PHASE_TEXT}>{userFriendlyConversationPhaseText}</Typography>
          <Box data-testid={DATA_TEST_ID.PROGRESS_BAR_LABEL}>{percentageInText}</Box>
        </Box>
      </Box>
    </Box>
  );
};

export default ChatProgressBar;
