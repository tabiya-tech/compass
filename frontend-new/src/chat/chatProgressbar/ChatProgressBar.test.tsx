import { render, screen } from "src/_test_utilities/test-utils";
import { ConversationPhase } from "./types";
import ChatProgressBar, { DATA_TEST_ID } from "./ChatProgressBar";
import * as GetUserFriendlyConversationPhaseNameModule from "./getUserFriendlyConversationPhaseName";
import { waitFor } from "@testing-library/react";

describe("ChatProgresBar", () => {
  describe("render tests", () => {
    it("should render the progress bar with the correct phase and percentage", async () => {
      // GIVEN a percentage and phase
      const givenPercentage = 75;
      const givenPhase = ConversationPhase.INTRO;

      // AND getUserFriendlyConversationPhaseName is mocked
      const givenReturnedUserFriendlyPhaseName = "given-returned-user-friendly-phase-name";
      const mockedGetUserFriendlyConversationPhase =
        jest.spyOn(GetUserFriendlyConversationPhaseNameModule, "getUserFriendlyConversationPhaseName")
          .mockReturnValue(givenReturnedUserFriendlyPhaseName)

      // WHEN the ChatProgressBar is rendered
      render(<ChatProgressBar percentage={givenPercentage} phase={givenPhase} current={null} total={null} />);

      // THEN the getUserFriendlyConversationPhaseName function should be called with the correct phase.
      expect(mockedGetUserFriendlyConversationPhase).toHaveBeenCalledWith({
        phase: givenPhase,
        percentage: givenPercentage,
        current: null,
        total: null,
      });

      // AND the container should be in the document
      const container = screen.getByTestId(DATA_TEST_ID.CONTAINER);

      // AND the progress bar text should be in the document
      const progressBar = screen.getByTestId(DATA_TEST_ID.PROGRESS_BAR_PHASE_TEXT);
      expect(progressBar).toBeInTheDocument();
      expect(progressBar).toHaveTextContent(givenReturnedUserFriendlyPhaseName);

      // AND the progress bar label should be in the document
      const progressBarLabel = screen.getByTestId(DATA_TEST_ID.PROGRESS_BAR_LABEL);
      expect(progressBarLabel).toBeInTheDocument();
      expect(progressBarLabel).toHaveTextContent(`${givenPercentage}%`);

      // AND the width of the progress bar should be set to the percentage
      // Wait for it because of the animation.

      const progressBarElement = screen.getByTestId(DATA_TEST_ID.PROGRESS_BAR);
      expect(progressBarElement).toBeInTheDocument();
      await waitFor(() => {
        expect(progressBarElement).toHaveStyle(`width: ${givenPercentage}%`);
      })

      // AND container should match snapshot.
      expect(container).toMatchSnapshot();
    })
  })
});
