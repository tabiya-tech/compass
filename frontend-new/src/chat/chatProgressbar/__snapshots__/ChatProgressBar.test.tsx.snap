// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatProgresBar render tests should render the progress bar with the correct phase and percentage 1`] = `
<div
  class="MuiBox-root css-156iqjm"
  data-testid="chat-progress-bar-container-2a76494f-351d-409d-ba58-e1b2cfaf2a53"
>
  <div
    data-testid="chat-progress-bar-2a76494f-351d-409d-ba58-e1b2cfaf2a53"
    style="background-color: rgb(0, 255, 145); height: 100%; grid-row: 1; border-radius: 4px; grid-column: 1; width: 75%;"
  />
  <div
    class="MuiBox-root css-1d3l42j"
  >
    <span
      class="MuiTypography-root MuiTypography-progressBarText css-110x8ue-MuiTypography-root"
      data-testid="chat-progress-bar-text-2a76494f-351d-409d-ba58-e1b2cfaf2a53"
    >
      given-returned-user-friendly-phase-name
    </span>
    <div
      class="MuiBox-root css-0"
      data-testid="chat-progress-bar-label-2a76494f-351d-409d-ba58-e1b2cfaf2a53"
    >
      75%
    </div>
  </div>
</div>
`;
